service/history/historyBuilder.go

```diff
@@ -477,6 +479,18 @@ func (b *historyBuilder) newWorkflowExecutionStartedEvent(
 	attributes.ChildPolicy = request.ChildPolicy
 	attributes.ContinuedExecutionRunId = previousRunID
 	attributes.Identity = common.StringPtr(common.StringDefault(request.Identity))
+	attributes.RetryPolicy = request.RetryPolicy
+	attributes.Attempt = common.Int32Ptr(startRequest.GetAttempt())
```

Comment:
We should allow to have Attempt and expiration being nil.

Reply:
yes, if RetryPolicy is nil, it will just set the nil to attributes. The GetAttempt() will return 0 for nil case.